### ISSUE TITLE
Certain F# libraries doesn't work with Central Package Management

### ISSUE THREAD

**Issue by ljb** [NuGet/Home#12816]
### NuGet Product Used

dotnet.exe

### Product Version

Dotnet 7.0.307

### Impact

Other

### Repro Steps & Context

Enabling Central Package Management doesn't work with certain F# libraries.

The following libraries are known to be problematic:
* `Aether`
* `FParsec`
* `Chiron` (depends on `FParsec`)

The following is a minimal example

`Directory.Packages.props`:
```
<Project>

  <PropertyGroup>
    <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
  </PropertyGroup>

  <ItemGroup>
    <PackageVersion Include="Aether" Version="8.3.1" />
    <PackageVersion Include="Chiron" Version="6.3.1" />
    <PackageVersion Include="FParsec" Version="1.1.1" />
    <PackageVersion Include="FSharp.Data" Version="6.2.0" />
  </ItemGroup>

</Project>
```

`test.fsproj`:
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net7.0</TargetFramework>
    <LangVersion>7.0</LangVersion>
  </PropertyGroup>

  <ItemGroup>
    <!--<PackageReference Include="FSharp.Data" />-->
    <!-- This has also been tried for Chiron and Aether -->
    <PackageReference Include="FParsec" />
  </ItemGroup>

  <ItemGroup>
    <Compile Include="Library.fs" />
  </ItemGroup>

</Project>
```

`Library.fs`:
```
namespace test

module MyLibrary =
    let print () =
        let world = "world"
        printf $"Hello %s{world}"
```

Running `dotnet build` will result in the following error:
```
C:\repos\test\Library.fs(6,16): error FS3353: Feature 'string interpolation' requires the F# library for language version 5.0 or greater. [C:\repos\test\test.fsproj]
```

Setting the `LangVersion` manually doesn't solve the issue.

Strangely enough, if a dependency is added to `FSharp.Data` it starts working. However, I have a much larger project that depends on both the problematic libraries and `FSharp.Data` and that doesn't work.

If I remove the `Directory.Packages.props` file and set the `PackageReference` with an explicit version like this, `<PackageReference Include="FParsec" Version="1.1.1" />`, it works as expected.

I'm not totally sure if this is a bug in NuGet, the DotNet SDK or in the F# compiler, so I apologize if I have posted this to the wrong place.

**Comment by ljb** [comment]
Adding an explicit reference to `FSharp.Core` solves the issue for all the projects in my solution.

So I guess that the problem with the mentioned libraries are that they are old, which results in my code also using an old version of `FSharp.Core`.

I know that the issue https://github.com/NuGet/Home/issues/11949 exists which mentions problems with `FSharp.Core` being loaded implicitly. I though my problem was unrelated to that, but maybe not.

Just having a `PackageVersion` for `FSharp.Core` wasn't it enough, which caught me by surprise. Previously it was enough to have something like `<PackageReference Update="FSharp.Core" Version=7.0.300 />` in `Directory.Build.targets`, but with Central Package Management I need both the `PackageVersion` and a `PackageReference` in each F#-project.

Maybe this issue can be closed if the recommended approach is to do what I describe above.

**Comment by zivkan** [comment]
@ljb the last comment in #11949 has a link to:

* https://github.com/dotnet/fsharp/pull/13920

So, when not using CPM, the F# SDK automatically adds a reference to the package, but when CPM is enabled, it does not reference those packages. 

Since this is specific to F# projects, its SDK, and the F# language itself, I need to defer to the F# team on this.

**Comment by kartheekp-ms** [comment]
@ljb - Considering the context and insights from the discussions above, it seems this issue is more closely related to the F# SDK and the F# language itself. Given this, would it be appropriate to close this issue here and move the discussion to the [dotnet/fsharp](https://github.com/dotnet/fsharp) repository?

Regarding your observation:

> Just having a PackageVersion for FSharp.Core wasn't it enough, which caught me by surprise. Previously it was enough to have something like \<PackageReference Update="FSharp.Core" Version=7.0.300 \/> in Directory.Build.targets, but with Central Package Management I need both the PackageVersion and a PackageReference in each F#-project.

Central Package Management (CPM) is designed to help users manage package versions centrally. Not all projects will necessarily reference the same packages, which explains the need to specify package references in individual project files.

That said, I have a query to further understand the situation: Have you tried using `<PackageReference Update="FSharp.Core" />` within the `Directory.Build.targets` file with CPM enabled? If so, does this configuration apply to all projects in the respective folder or subfolder? Essentially, I'm curious if every project under the Directory.Build.targets folder is then referencing the `FSharp.Core` package.

**Comment by ljb** [comment]
@kartheekp-ms:

For me it's okay to close this issue. I have found an acceptable workaround, and the issue can probably be of help for other people having trouble with F# and CPM.

I tried setting `<PackageReference Update="FSharp.Core" />` in `Directory.Build.targets`, but that didn't help. I still need to set `<PackageReference Include="FSharp.Core" />` explicitly in my F# projects.

I found an existing issue in dotnet/fsharp that is quite similar:
https://github.com/dotnet/fsharp/issues/14823

So it seems the F# developers are aware of this kind of problem.
